src/p2f.c

```diff
@@ -1312,17 +1344,27 @@ static void flow_record_print_json
      */
     zprintf(ctx->output, "{");
 
-    if (ipv4_addr_needs_anonymization(&rec->key.sa)) {
-        zprintf(ctx->output, "\"sa\":\"%s\",", addr_get_anon_hexstring(&rec->key.sa));
-    } else {
-        inet_ntop(AF_INET, &rec->key.sa, ipv4_addr, INET_ADDRSTRLEN);
-        zprintf(ctx->output, "\"sa\":\"%s\",", ipv4_addr);
-    }
-    if (ipv4_addr_needs_anonymization(&rec->key.da)) {
-        zprintf(ctx->output, "\"da\":\"%s\",", addr_get_anon_hexstring(&rec->key.da));
+    if (rec->ip_type == ETH_TYPE_IPV6) {
+        inet_ntop(AF_INET6, &rec->key.sa.v6_sa, ipv6_addr, INET6_ADDRSTRLEN);
+        zprintf(ctx->output, "\"sa\":\"%s\",", ipv6_addr);
+        inet_ntop(AF_INET6, &rec->key.da.v6_da, ipv6_addr, INET6_ADDRSTRLEN);
+        zprintf(ctx->output, "\"da\":\"%s\",", ipv6_addr);
     } else {
-        inet_ntop(AF_INET, &rec->key.da, ipv4_addr, INET_ADDRSTRLEN);
-        zprintf(ctx->output, "\"da\":\"%s\",", ipv4_addr);
+        char buffer[33];
```

Comment:
Should we use IPV4_ANON_LEN instead of 33?

Reply:
yes, I will change